Lib/asyncio/locks.py

```diff
@@ -378,20 +383,18 @@ async def acquire(self):
         fut = self._get_loop().create_future()
         self._waiters.append(fut)
 
-        # Finally block should be called before the CancelledError
-        # handling as we don't want CancelledError to call
-        # _wake_up_first() and attempt to wake up itself.
         try:
-            try:
-                await fut
-            finally:
-                self._waiters.remove(fut)
-        except exceptions.CancelledError:
-            if not fut.cancelled():
+            await fut
```

Comment:
Same thing as before -- I actually like the nested `try`/`finally` phrasing better. Also, again, sneaky about catching all errors.

Reply:
Fair point, especially if it costs nothing.